index.html

```diff
@@ -11872,9 +11872,14 @@ <h2>Excluding Elements from the Accessibility Tree</h2>
 	  <h2>Including Elements in the Accessibility Tree</h2>
           <p>If not already excluded from the accessibility tree per the rules above in <a href="#tree_exclusion">Excluding Elements in the Accessibility Tree</a>, user agents MUST provide an <a>accessible object</a> in the <a>accessibility tree</a> for <abbr title="Document Object Model">DOM</abbr> <a>elements</a> that meet any of the following criteria:</p>
           <ul>
-            <li>Text nodes</li>
-            <li>Elements that may fire an <a>accessibility <abbr title="Application Program Interface">API</abbr></a> <a>event</a>.</li>
-            <li>Elements that are focusable, or have an ID <a>attribute</a> and an ancestor with the <pref>aria-activedescendant</pref> attribute that matches the implicit or explicit <a>semantics</a> of the required context role. In either case, the element may receive focus and may fire an <a>accessibility <abbr title="Application Program Interface">API</abbr></a> focus event.</li>
+            <li>Text nodes.</li>
+            <li>Elements that may fire an <a>accessibility <abbr title="Application Program Interface">API</abbr></a> <a>event</a>.
+              <ul>
+                <li>Elements that are focusable may receive focus and may fire an <a>accessibility <abbr title="Application Program Interface">API</abbr></a> focus event.</li>
+                <li>Elements that have an ID <a>attribute</a>, and are owned by or have an ancestor with the <pref>aria-activedescendant</pref> attribute that matches the ID, may receive focus and may fire an <a>accessibility <abbr title="Application Program Interface">API</abbr></a> focus event.</li>
```

Comment:
In ARIA 1.1, we allow aria-activedescendant on a textbox to point to a descendant of an element controlled by the textbox. This was a critical change for the combobox pattern. So, this text should probably change to reflect that as well. Generally, those elements would probably in the tree for another reason, but we probably can't assume that is always the case.

Reply:
Can use some of the words in aria-activedescendant.

Reply:
Concern: I'm not certain these changes are readily implementable in a performant manner.

For example, it's achievable to update the tree on events, such as focus... so if an otherwise inaccessible element were focused, it could become accessible at that time. Several implementations do this today. However, it's not as straightforward to determine what may or may not trigger some relevant event at some point in the future. Likewise, it's not clear how to expose some elements referenced by ARIA relationship attributes. It's reasonable to calculate a label from `aria-labelledby` referencing a hidden element, but exposing a hidden element referenced by `aria-controls` or `aria-owns` would be unexpected. I'm not convinced it's what authors or users would want.

One example: most combobox implementations use both `aria-owns` for the menu relationship and `aria-activedescendant` for the selected menu item. The visibility of this "popup" is usually controlled by setting `display:none` on the menu when it is irrelevant or not displayed. I've never seen an implementation that removes the `aria-owns` relationship each time the menu is shown or hidden. Authors usually leave these attributes in place.

This spec language seems to indicate that most combobox implementations should always expose the sub menu, even if it is not displayed. I don't believe anyone actually wants that.

Minor nit: Every use of the term "owned by" should link to the definition.